cpp/oneapi/dal/algo/pca/backend/cpu/train_kernel_svd.cpp

```diff
@@ -51,64 +51,103 @@ template <typename Float>
+
+    if (desc.get_normalization_mode() == normalization::mean_center) {
+        norm_alg->parameter().doScale = false;
+        daal_pca_parameter.doScale = false;
+    }
+
+    daal_pca_parameter.isDeterministic = desc.get_deterministic();
+    daal_pca_parameter.normalization = norm_alg;
+    daal_pca_parameter.resultsToCompute =
         std::uint64_t(daal_pca::mean | daal_pca::variance | daal_pca::eigenvalue);
 
-    interop::status_to_exception(
-        interop::call_daal_kernel<Float, daal_pca_svd_kernel_t>(ctx,
-                                                                dtype,
-                                                                *daal_data.get(),
-                                                                &parameter,
-                                                                *daal_eigenvalues.get(),
-                                                                *daal_eigenvectors.get(),
-                                                                *daal_means.get(),
-                                                                *daal_variances.get()));
+    interop::status_to_exception(interop::call_daal_kernel<Float, daal_pca_svd_kernel_t>(
+        ctx,
+        dtype,
+        *daal_data,
+        *daal_eigenvectors,
+        *daal_singular_values,
+        *daal_means,
+        *daal_variances,
+        daal_eigenvalues.get(),
+        daal_explained_variances_ratio.get(),
+        &daal_pca_parameter));
+
+    model_t model;
 
     if (desc.get_result_options().test(result_options::eigenvectors)) {
-        const auto mdl = model_t{}.set_eigenvectors(
-            homogen_table::wrap(arr_eigvec, component_count, column_count));
-        result.set_model(mdl);
+        model.set_eigenvectors(homogen_table::wrap(arr_eigvec, component_count, column_count));
+    }
+
+    if (desc.get_result_options().test(result_options::singular_values)) {
+        result.set_singular_values(homogen_table::wrap(arr_singular_values, 1, component_count));
     }
 
     if (desc.get_result_options().test(result_options::eigenvalues)) {
         result.set_eigenvalues(homogen_table::wrap(arr_eigval, 1, component_count));
+        model.set_eigenvalues(homogen_table::wrap(arr_eigval, 1, component_count));
+    }
+
+    if (desc.get_result_options().test(result_options::explained_variances_ratio)) {
+        result.set_explained_variances_ratio(
+            homogen_table::wrap(arr_explained_variances_ratio, 1, component_count));
     }
     if (desc.get_result_options().test(result_options::vars)) {
         result.set_variances(homogen_table::wrap(arr_vars, 1, column_count));
+        model.set_variances(homogen_table::wrap(arr_vars, 1, column_count));
     }
     if (desc.get_result_options().test(result_options::means)) {
         result.set_means(homogen_table::wrap(arr_means, 1, column_count));
+        model.set_means(homogen_table::wrap(arr_means, 1, column_count));
     }
 
+    result.set_model(model);
+
```

Comment:
Output code is also duplicated. PLease isolate the common part into a separate function